<a href="https://colab.research.google.com/github/geoshibu2000/Python-Data-Analytics-project./blob/main/PDA_project_GeoShibuMathew.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
df = pd.DataFrame(pd.read_csv('/content/train (1).csv'))

In [ ]:
new_file = df.isnull().sum()
new_file

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [ ]:
x = new_file[new_file> (40/100 * df.shape[0])]
x

Cabin    687
dtype: int64

In [ ]:
df.shape

(891, 12)

In [ ]:
df.size

10692

In [ ]:
df.fillna(df.mean(), inplace=True)

In [ ]:
df['Embarked'].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

In [ ]:
df.groupby(['Embarked'])['Survived'].mean()

Embarked
C    0.553571
Q    0.389610
S    0.336957
Name: Survived, dtype: float64

In [ ]:
df.groupby(['Pclass'])['Survived'].mean()

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

In [ ]:
df['familysize'] = df['SibSp']+df['Parch']


In [ ]:
df.drop(['SibSp','Parch'], axis=1, inplace=True)

In [ ]:
df['Alone'] = [0 if df['familysize'][i]>0 else 1 for i in df.index]
df.head()

In [ ]:
df.groupby(['Alone'])['Survived'].mean()

Alone
0    0.505650
1    0.303538
Name: Survived, dtype: float64

In [ ]:
df['Sex'] = [0 if df['Sex'][i]=='male' else 1 for i in df.index]


In [ ]:
df.groupby(['Sex'])['Survived'].mean()

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

In [ ]:
nf = pd.DataFrame(df, columns=['Name'])
nf

In [ ]:
df.dropna(axis=1, inplace=True)
df

In [ ]:
df.tail(5)

In [ ]:
df.isnull().any()

Conclusions -

1. Most of the passengers did not have Cabins. 
2. The passengers who were not alone had a higher survival rate. 
3. The female passengers had a higher survival rate than men. 
4. The passengers who boarded the ship at Cherbourg station, survived more in proportion than the others. 
5. The passengers who were allotted higher Pclass tickets, had a higher priority and thereby had a higher survival rate than the others.
6. The DataFrame had 891 rows, 12 columns, and 10692 number of elements initially.
